### ISSUE TITLE
getLabelProps and accessibility

### ISSUE THREAD

**Issue by markau** [techniq/mui-downshift#51]
This is a really nice library, thank you.

I am seeing some accessibility errors (Missing form label, broken ARIA reference) and I guess I do not know how to use getLabelProps. In the base downshift library it seems straightforward but I start to get lost with the extra layer of abstraction. It seems that `getRootProps `should be getting the `downshiftProps`, which should contain the getLabelProps(). I don't know how to implement that though ... I'd appreciate a hint!

**Comment by techniq** [comment]
Thanks!  I've not done much accessibility work (and would appreciate PR's if we can make improvements here), but I think [this](https://github.com/techniq/mui-downshift/blob/master/src/Input.js#L38) might be what you're looking for:

```js
<StarWarsSelect
      getInputProps={dowshiftProps => ({
        label: 'Star Wars character',
        labelProps: { /* ... */ }
      })}
    />
```

We might need to call `downshift.getLabelProps()` [here](https://github.com/techniq/mui-downshift/blob/master/src/Input.js#L45) as well.

**Comment by markau** [comment]
Hey thanks for the reply. I only saw 4 accessibility issues, so am keen to explore and better understand the code (both yours and in the base downshift library) and hopefully offer suggestions. I'll open new issues with anything specific.

**Comment by markau** [comment]
You are correct that `downshift.getLabelProps()` was needed, in order to find the label applied via the `label `property in mui-downshift `getInputProps`. 

The `aria-label` properties on the buttons are required because the buttons themselves have no text, and assistive devices like screenreaders need to read something out.

I am only using an implementation in the "basic" story, but these small changes mean this is now AAA compliant which is a testament to material-ui, downshift, and the way you have brought them together. For info, you can use the [WAVE Chrome extension](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh) for accessibility testing (although it doesn't work very well with the storybooks embedding).

**Comment by techniq** [comment]
@markau Thanks for the feedback and PR, I really appreciate it.  I've published your changes as version `1.2.0` if you want to take a look and let me know.
